### PR TITLE
#13 Post Backend CRUD routes

### PR DESCRIPTION
**Pull request by loren589**
Added all the backend routes related to the posts. I also added the post picture attribute to the post route, but I have some questions as to how picture files get uploaded to the database, which I'll ask in the meeting on Tuesday.

### CODE REVIEW

File: routes/api/posts.js [hunk 1]
```diff
@@ -1,9 +1,198 @@
...
+
+// @route POST api/posts
+// @desc creating a post
+// @access Private
+router.post('/', [auth, check('text', "Text is necessary").not().isEmpty()], //do we actually want the post text to be necessary? I did remove it from the Post Model...
+async (req,res) => {
+    const errors= validationResult(req);
+    if(!errors.isEmpty()) {//now doing error checks below -->using try catch for server error
+        return res.status(400).json({errors: errors.array() });
+    }
```
Comment (gaticosalas) [inline routes/api/posts.js:22]: Text will not be necessary but picture will, so just change the check method to:
`check('picture', 'Picture is necessary').not().Empty()`.

File: routes/api/posts.js [hunk 2]
```diff
@@ -1,9 +1,198 @@
...
+
+    try {//user is logged in, has token, and thus ID is available
+        const user = await (await User.findById(req.user.id)).select('-password'); //don't need to send in the password so omit it
+        const newPost = new Post({
+            picture: req.body.string, //look into using Multer? It's a node.js middleware used to upload files to the DB
```
Comment (gaticosalas) [inline routes/api/posts.js:27]: You're right about Multer! checking into it!

File: routes/api/posts.js [hunk 3]
```diff
@@ -1,9 +1,198 @@
...
+            text: req.body.text,
```
Comment (gaticosalas) [inline routes/api/posts.js:28]: Reminder that you called the text 'caption' in your Schema, so rename this to caption too to stay consistent!

**Review by gaticosalas — CHANGES_REQUESTED** [review]
Great job, just a few little fixes!

**Review by gaticosalas — APPROVED** [review]
Perfect!

**Comment by MKB-1** [comment]
Good Job!

**Review by MKB-1 — APPROVED** [review]

**Pull request merged**